lib/recurly/apple-pay.js

```diff
@@ -0,0 +1,225 @@
+import Promise from 'promise';
+import Emitter from 'component-emitter';
+import errors from '../errors';
+import {Pricing} from './pricing';
+
+const debug = require('debug')('recurly:apple-pay');
+
+const SESSION_TIMEOUT = 300000; // 5m
+
+export function factory (options) {
+  return new ApplePay(Object.assign({}, options, { recurly: this }));
+};
+
+/**
+ * Initializes an Apple Pay session.
+ *
+ * @param {Object} options
+ * @constructor
+ * @public
+ */
+
+class ApplePay extends Emitter {
+  constructor (options) {
+    super();
+
+    this._ready = false;
+    this.config = {};
+    this.once('ready', () => this._ready = true);
+
+    // Detect whether Apple Pay is available
+    if (!ApplePaySession) {
+      this.initError = this.error('apple-pay-not-supported');
+    } else if (!ApplePaySession.canMakePayments()) {
+      this.initError = this.error('apple-pay-not-available');
+    }
+
+    if (!this.initError) {
+      this.configure(options);
+    }
+  }
+
+  get session () {
+    if (this._session) return this._session;
+
+    debug('Creating new Apple Pay session');
+
+    let session = new ApplePaySession(2, {
```

Comment:
Might be nice at some point to move this `2` to a constant somewhere so it's purpose is more obvious for future dev. I had to look it up as I forgot was it was for already.